### PR TITLE
[BFD-923] terraform cleanup and upgrade AWS provider

### PR DESCRIPTION
**Pull request by keithdadkins**
### Change Details
In preparation for adding an ECR module to our terraform, this PR:
- cleans up and formats our existing terrafrom codebase
- refactors our terraform to remove terraform validation warnings
- upgrades the AWS resource provider to v3 from v2

There are a lot of changes in this PR across a lot of files, but most of the changes have to deal with cleaning up code comments, spacing, etc.

Basically, I went through each .tf file, cleaned up dead code, removed superfluous code comments, and made a first pass at making our terraform more idiomatic. I then:
- Ran `terraform init -backend=false && terraform validate` in each environment and refactored all the bits that was causing warnings or errors.
- I then updated our AWS provider to the latest v3 version.
- I then ran `terraform fmt` on all .tf files
- And finally, I re-validated all environments once again to ensure there were no warnings, errors, or issues.

Once approved **but before merging**, I will _init_, _plan_, and _apply_ stateful resources in the test environment as well as deploy this PR to test (to ensure stateless resources are ok).

This has been a long time coming and will be a necessary part of upgrading our terraform from v0.12 to v0.13+.

Note: there is still a bunch of dead code, but we can tackle that at a later date.

### External References

- [BFD-923](https://jira.cms.gov/browse/BFD-923)

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Comment by keithdadkins** [comment]
Note: I merged master which included some minor terraform changes (rda stuff). I ran terraform validate in all environments again to ensure there were no breaking changes.

**Review by njdister — APPROVED** [review]
💥

File: ops/terraform/modules/stateless/main.tf [hunk 1]
```diff
@@ -11,7 +9,7 @@ locals {
   cw_period       = 60 # Seconds
   cw_eval_periods = 3
 
-  # Add new peerings here
+  # add new peerings here (#TODO: add dcgeo)
```
Comment (jzulim) [inline ops/terraform/modules/stateless/main.tf:12]: No needed. They leverage BCDA peer.

**Review by jzulim — APPROVED** [review]
Took the time to go line by line, looks great. I like the commenting changes.

**Review by JeremyIR — APPROVED** [review]

**Review by jzulim — APPROVED** [review]

File: insights/terraform/projects/ab2d/main.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ locals {
   tags      = {business = "OEDA", application = "bfd-insights", project="ab2d"}
   project   = "ab2d"
   database  = "ab2d"
-  partitions = [{name="dt", type="string", comment="Approximate delivery time"}]
+  partitions = [{name="dt", type=string, comment="Approximate delivery time"}]
```
Comment (jzulim) [inline insights/terraform/projects/ab2d/main.tf:5]: @nflearl this is the only change for AB2D to be aware of.

**Review by nflearl — APPROVED** [review]

**Pull request merged**